script/make_utils/run_use_case_examples.sh

```diff
@@ -1,129 +1,120 @@
 #!/usr/bin/env bash
 set -e
 
-DIR=$(dirname "$0")
-
-# shellcheck disable=SC1090,SC1091
-source "${DIR}/detect_docker.sh"
-
-if isDocker; then
-    echo "Can not run in docker -> this script needs to install new virtualenvs"
-    exit 1
-fi
-
-
-CML_DIR=$(pwd)
-USE_CASE_REL_DIR="use_case_examples"
-USE_CASE_DIR="${CML_DIR}/${USE_CASE_REL_DIR}"
-
-if [ ! -d "$USE_CASE_DIR" ]; then
-    echo "This script must be run in the Concrete ML source root where the '$USE_CASE_REL_DIR' directory is present"
-    exit 1
-fi
-
-echo "Refreshing notebooks with PIP installed Concrete ML"
-
-# shellcheck disable=SC2143
-if [[ $(git ls-files --others --exclude-standard | grep ${USE_CASE_REL_DIR}) ]]; then
-    echo "This script must be run in a clean clone of the Concrete ML repo"
-    echo "This directory has untracked files in ${USE_CASE_REL_DIR}"
-    echo "You can LIST all untracked files using: "
-    echo
-    # shellcheck disable=SC2028
-    echo "   git ls-files --others --exclude-standard | grep ${USE_CASE_REL_DIR}"
-    echo
-    echo "You can REMOVE all untracked files using: "
-    echo
-    # shellcheck disable=SC2028
-    echo "   git ls-files --others --exclude-standard | grep ${USE_CASE_REL_DIR} | xargs -0 -d '\n' --no-run-if-empty rm"
-    echo
-    exit 1
-fi
-
-if [[ -z "${USE_CASE}" ]]; then
-  # shellcheck disable=SC2207
-  LIST_OF_USE_CASES=($(find "$USE_CASE_DIR/" -mindepth 1 -maxdepth 2 -type d | grep -v checkpoints))
-else
-  LIST_OF_USE_CASES=("${USE_CASE}")
-  if [ ! -d "${USE_CASE}" ]; then 
-    echo "The use case specified to be executed, ${USE_CASE}, does not exist"
-    exit 1
-  fi
-fi
-
-if [ ! "$(docker images -q zamafhe/concrete-ml:latest 2> /dev/null)" ]; then
-    # BUILD THE DOCKER IMAGE
-    echo "Building docker image"
-    poetry build && mkdir -p pkg && cp dist/* pkg/ && make release_docker
-    docker tag concrete-ml-release:latest zamafhe/concrete-ml:latest
-fi
-
-# shellcheck disable=SC2068
-for EXAMPLE in ${LIST_OF_USE_CASES[@]}
-do
-    EXAMPLE_NAME=$(basename "${EXAMPLE}")
-
-    if [ -f "${EXAMPLE}/Makefile" ]; then
-        echo "*** Processing example ${EXAMPLE_NAME}"
-    else
-        continue
+CURRENT_DIR=$(pwd)
+USE_CASE_DIR_NAME="use_case_examples"
+USE_CASE_DIR="${CURRENT_DIR}/${USE_CASE_DIR_NAME}"
+
+export USE_CASE_DIR  # Required for the Makefile of the use case examples
+
+check_directory_exists() {
```

Comment:
ah also, could you just add some comments in this file, just to make things clear. Even if error messages and function names are pretty clear, would be great to just explain a bit what each function (as well as some of their crucial steps) is for !

Reply:
Added a onel ine comment above each function